### PR TITLE
🐛 Encrypt exclude_tags parameter

### PR DESCRIPTION
**Pull request by LRueckert**
### Summary of the issue
- as discussed in slack: `exclude_tags` need to be encrypted
- I also opened a PR in phdp-records to make the tags be combined with `or` (merged now)

### How to reproduce your bugfix or feature
Describe how to inspect and verify your changes.

### Due diligence checklist
- [x] Updated version in package.json if applicable.
- [x] Added/updated tests.
- [ ] Add documentation
- [ ] If this change affects SDK consumers: communicate changes.

### TODOs
- [ ] Mention any outstanding issues or tasks you need to perform after the merge.

### CODE REVIEW

File: src/services/recordService.ts [hunk 1]
```diff
@@ -205,16 +206,19 @@ const recordService = {
...
-                tags,
-              })
-            );
-          }
-          return params;
+          const encryptedTagsPromise = params?.tags?.length
+            ? Promise.all(params.tags.map(tag => symEncryptString(user.tek, tag)))
+            : Promise.resolve([]);
+          const encryptedExceptTagsPromise = params?.exclude_tags?.length
+            ? Promise.all(params.exclude_tags.map(tag => symEncryptString(user.tek, tag)))
```
Comment (Burtchen) [inline src/services/recordService.ts:213]: Can we rename the map'ed tag param to exludedTag?

File: src/services/recordService.ts [hunk 2]
```diff
@@ -205,16 +206,19 @@ const recordService = {
...
-                ...params,
-                tags,
-              })
-            );
-          }
-          return params;
+          const encryptedTagsPromise = params?.tags?.length
+            ? Promise.all(params.tags.map(tag => symEncryptString(user.tek, tag)))
+            : Promise.resolve([]);
+          const encryptedExceptTagsPromise = params?.exclude_tags?.length
```
Comment (Burtchen) [inline src/services/recordService.ts:212]: Maybe `excludedTagsPromise`?

File: src/services/fhirService.ts [hunk 3]
```diff
@@ -381,6 +383,16 @@ export const prepareSearchParameters = (params: IParams) => {
...
 
+  if (params.exclude_tags) {
+    const excludeTags = taggingUtils.generateCustomTags(params.exclude_tags);
+    parameters.exclude_tags = excludeTags;
+  }
+
+  if (params.exclude_flags) {
+    parameters.exclude_tags = [...new Set([...(params.exclude_tags || []), ...params.exclude_flags])];
+    delete parameters.exclude_flags
+  }
```
Comment (LRueckert) [inline src/services/fhirService.ts:394]: The `flag=appdata` that was passed along for count and fetchResources was being treated as a custom tag, therefore making it `custom=flag%3Dappdata`, which then obviously didn't work for filtering.

My solution to this is adding additional `exclude_flags`, which just get appended after the custom tags are handled.

File: src/services/fhirService.ts [hunk 4]
```diff
@@ -381,6 +383,18 @@ export const prepareSearchParameters = (params: IParams) => {
...
   }
 
+  if (params.exclude_tags) {
+    const excludeTags = taggingUtils.generateCustomTags(params.exclude_tags);
+    parameters.exclude_tags = excludeTags;
+  }
+
+  if (params.exclude_flags) {
+    parameters.exclude_tags = [
+      ...new Set([...(parameters.exclude_tags || []), ...params.exclude_flags]),
```
Comment (Burtchen) [inline src/services/fhirService.ts:393]: Note to self: lodash uniq this in a future update :trollface:

**Pull request merged**